spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -0,0 +1,566 @@
+        // can return any DataFrame.
+        return DeltaSharingScanUtils.internalCreateDataFrame(spark, schema)
+      }
+      (startOffset.tableVersion, startOffset.index, startOffset.isStartingVersion,
+        Some(startOffset.sourceVersion))
+    }
+    logDebug(s"start: $startOffsetOption end: $end")
+
+    // Proceed the offset as the files before the endOffset are processed.
+    previousOffset = endOffset
+    val createdDf = getFileChangesAndCreateDataFrame(
+      startVersion, startIndex, isStartingVersion, endOffset
+    )
+
+    createdDf
+  }
+
+  override def stop(): Unit = {}
+
+  override def toString(): String = s"DeltaSharingSource[${deltaLog.table.toString}]"
+
+  /**
+   * Class that helps controlling how much data should be processed by a single micro-batch.
+   */
+  class AdmissionLimits(
+    maxFiles: Option[Int] = options.maxFilesPerTrigger,
+    var bytesToTake: Long = options.maxBytesPerTrigger.getOrElse(Long.MaxValue)
+  ) {
+
+    var filesToTake = maxFiles.getOrElse {
+      if (options.maxBytesPerTrigger.isEmpty) {
+        DeltaSharingOptions.MAX_FILES_PER_TRIGGER_OPTION_DEFAULT
+      } else {
+        Int.MaxValue - 8 // - 8 to prevent JVM Array allocation OOM
+      }
+    }
+
+    def toReadLimit: ReadLimit = {
+      if (options.maxFilesPerTrigger.isDefined && options.maxBytesPerTrigger.isDefined) {
+        CompositeLimit(
+          ReadMaxBytes(options.maxBytesPerTrigger.get),
+          ReadLimit.maxFiles(options.maxFilesPerTrigger.get).asInstanceOf[ReadMaxFiles])
+      } else if (options.maxBytesPerTrigger.isDefined) {
+        ReadMaxBytes(options.maxBytesPerTrigger.get)
+      } else {
+        ReadLimit.maxFiles(
+          options.maxFilesPerTrigger.getOrElse(
+            DeltaSharingOptions.MAX_FILES_PER_TRIGGER_OPTION_DEFAULT))
+      }
+    }
+
+    /** Whether to admit the next file */
+    def admit(addFile: Option[AddFile]): Boolean = {
+      if (addFile.isEmpty) return true
+      val shouldAdmit = filesToTake > 0 && bytesToTake > 0
+      filesToTake -= 1
+
+      bytesToTake -= addFile.get.size
+      shouldAdmit
+    }
+  }
+
+  object AdmissionLimits {
+    def apply(limit: ReadLimit): Option[AdmissionLimits] = limit match {
+      case _: ReadAllAvailable => None
+      case maxFiles: ReadMaxFiles => Some(new AdmissionLimits(Some(maxFiles.maxFiles())))
+      case maxBytes: ReadMaxBytes => Some(new AdmissionLimits(None, maxBytes.maxBytes))
+      case composite: CompositeLimit =>
+        Some(new AdmissionLimits(Some(composite.files.maxFiles()), composite.bytes.maxBytes))
+      case other => throw DeltaSharingErrors.unknownReadLimit(other.toString())
+    }
+  }
+
+  /**
+   * Extracts whether users provided the option to time travel a relation. If a query restarts from
+   * a checkpoint and the checkpoint has recorded the offset, this method should never been called.
+   */
+  protected lazy val getStartingVersion: Option[Long] = {
```

Comment:
can this be private?
Same question for other protected methods.

Reply:
changed all of them to private.